src/managers/input/InputManager.cpp

```diff
@@ -240,6 +240,15 @@ void CInputManager::mouseMoveUnified(uint32_t time, bool refocus) {
     if (!foundSurface)
         foundSurface = g_pCompositor->vectorToLayerSurface(mouseCoords, &PMONITOR->m_aLayerSurfaceLayers[ZWLR_LAYER_SHELL_V1_LAYER_TOP], &surfaceCoords, &pFoundLayerSurface);
 
+    // then IME popups
+    if (!foundSurface) {
```

Comment:
according to how it's rendered, popups are above TOP layers, and below OVERLAY ones. This should be moved a bit up.